packages/mask/src/plugins/Avatar/hooks/usePersonas.ts

```diff
@@ -1,39 +1,30 @@
 import { NextIDPlatform } from '@masknet/shared-base'
 import { NextIDProof } from '@masknet/web3-providers'
-import { first } from 'lodash-unified'
 import { useAsyncRetry } from 'react-use'
 import { useSubscription } from 'use-subscription'
 import { usePersonaConnectStatus } from '../../../components/DataSource/usePersonaConnectStatus'
 import { activatedSocialNetworkUI } from '../../../social-network'
-import { sortPersonaBindings } from '../../../utils'
 import { context } from '../context'
 
 export function usePersonas(userId?: string) {
     const personaConnectStatus = usePersonaConnectStatus()
-    const currentIdentifier = useSubscription(context.currentVisitingProfile)
     const platform = activatedSocialNetworkUI.configuration.nextIDConfig?.platform as NextIDPlatform
     const identifier = useSubscription(context.lastRecognizedProfile)
+    const currentIdentifier = useSubscription(context.currentVisitingProfile)
     return useAsyncRetry(async () => {
-        if (!identifier?.identifier?.userId) return
-        const personaBindings = await NextIDProof.queryAllExistedBindingsByPlatform(
-            platform,
-            userId ?? identifier.identifier.userId,
-        )
+        if (!identifier?.identifier?.userId || !personaConnectStatus.currentPersona?.identifier.publicKeyAsHex) return
```

Comment:
Can we use `currentIdentifier.publicKey`?

Reply:
currentIdentifier  has no attribute publicKey